ddmrp/models/stock_buffer.py

```diff
@@ -1194,23 +1196,27 @@ def write(self, vals):
     @api.model
     def cron_ddmrp_adu(self, automatic=False):
         """calculate ADU for each DDMRP buffer. Called by cronjob."""
+        auto_commit = not getattr(threading.currentThread(), "testing", False)
         _logger.info("Start cron_ddmrp_adu.")
-        buffers = self.search([])
+        buffer_ids = self.search([]).ids
         i = 0
-        j = len(buffers)
-        for b in buffers:
-            try:
-                i += 1
-                _logger.debug("ddmrp cron_adu: {}. ({}/{})".format(b.name, i, j))
-                if automatic:
-                    with self.env.cr.savepoint():
+        j = len(buffer_ids)
+        for buffer_chunk_ids in split_every(50, buffer_ids):
+            for b in self.browse(buffer_chunk_ids):
+                try:
+                    i += 1
+                    _logger.debug("ddmrp cron_adu: {}. ({}/{})".format(b.name, i, j))
+                    if automatic:
+                        with self.env.cr.savepoint():
+                            b._calc_adu()
+                    else:
                         b._calc_adu()
-                else:
-                    b._calc_adu()
-            except Exception:
-                _logger.exception("Fail to compute ADU for buffer %s", b.name)
-                if not automatic:
-                    raise
+                    if auto_commit:
+                        self._cr.commit()
```

Comment:
Nitpick: Is it not better to put these two lines in a `else` clause of the `try/except` block?

Reply:
The commit is probably meant to be after one chunk, not after one record, for two reasons:

* the savepoint would be useless if we commit after each record
* in terms of efficiency, a commit after each record is a disaster, as the env is flushed and cleared, so all the records of the chunk are read again (if you have 10'000 buffers, you end up with 10'000 `SELECT` that read 50 ids each time)